easybuild/easyblocks/b/blender.py

```diff
@@ -0,0 +1,113 @@
+        return res[0]
+
+    def configure_step(self):
+        """Set CMake options for Blender"""
+        self.cfg['separate_build_dir'] = True
+
+        default_config_opts = {
+            'WITH_INSTALL_PORTABLE': 'OFF',
+            'WITH_BUILDINFO': 'OFF',
+            # disable SSE detection to give EasyBuild full control over optimization compiler flags being used
+            'WITH_CPU_SSE': 'OFF',
+            'CMAKE_C_FLAGS_RELEASE': '-DNDEBUG',
+            'CMAKE_CXX_FLAGS_RELEASE': '-DNDEBUG',
+            # these are needed unless extra dependencies are added for them to work
+            'WITH_GAMEENGINE': 'OFF',
+            'WITH_SYSTEM_GLEW': 'OFF',
+        }
+        for key in default_config_opts:
+            opt = '-D%s=' % key
+            if opt not in self.cfg['configopts']:
+                self.cfg.update('configopts', opt + default_config_opts[key])
+
+        # Python paths
+        python_root = get_software_root('Python')
+        if python_root:
+            shlib_ext = get_shared_lib_ext()
+            pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+            site_packages = os.path.join(python_root, 'lib', 'python%s' % pyshortver, 'site-packages')
+
+            numpy_root = self.find_glob_pattern(
+                os.path.join(site_packages, 'numpy-*-py%s-linux-x86_64.egg' % pyshortver))
+            requests_root = self.find_glob_pattern(os.path.join(site_packages, 'requests-*-py%s.egg' % pyshortver))
+            python_lib = self.find_glob_pattern(
+                os.path.join(python_root, 'lib', 'libpython%s*.%s' % (pyshortver, shlib_ext)))
+            python_include_dir = self.find_glob_pattern(os.path.join(python_root, 'include', 'python%s*' % pyshortver))
+
+            self.cfg.update('configopts', '-DPYTHON_VERSION=%s' % pyshortver)
+            self.cfg.update('configopts', '-DPYTHON_LIBRARY=%s' % python_lib)
+            self.cfg.update('configopts', '-DPYTHON_INCLUDE_DIR=%s' % python_include_dir)
+            self.cfg.update('configopts', '-DPYTHON_NUMPY_PATH=%s' % numpy_root)
+            self.cfg.update('configopts', '-DPYTHON_REQUESTS_PATH=%s' % requests_root)
+
+        # OpenEXR
+        openexr_root = get_software_root('OpenEXR')
+        if openexr_root:
+            self.cfg.update('configopts', '-DOPENEXR_INCLUDE_DIR=%s' % os.path.join(openexr_root, 'include'))
+
+        # OpenColorIO
+        if get_software_root('OpenColorIO'):
+            self.cfg.update('configopts', '-DWITH_OPENCOLORIO=ON')
+
+        # CUDA
+        if get_software_root('CUDA'):
+            self.cfg.update('configopts', '-DWITH_CYCLES_CUDA_BINARIES=ON')
+
+        super(EB_Blender, self).configure_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for Blender."""
+
+        custom_paths = {
+            'files': ['bin/blender'],
+            'dirs': [],
+        }
+
```

Comment:
empty lines above & below can be dropped imho (but that's just personal preference)